### DISCUSSION TITLE
Does Analysis Template not support HTTPS endpoints for prometheus?

### DISCUSSION THREAD

**Discussion by strada501** [argoproj/argo-rollouts#1332] [Q&A]
I used Prometheus query using AnalysisTemplate of argo rollouts, but when I set the HTTPS endpoint to address, the following error occurs.

 'Post "https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query": x509: certificate signed by unknown authority'

Does Analysis Template not support HTTPS endpoints?

**Comment by kostis-codefresh** [comment]
This is not an error specific to Argo Rollouts.

Argo Rollouts uses the standard Prometheus client behind the scenes to connect to Prometheus

https://github.com/prometheus/client_golang

We need to check if the client supports mTLS/insecure/self-signed certificates and possible expose the TLS setup the Argo Rollouts settings.

**Reply by NYKYM** [reply]
Hi, @kostis-codefresh 

May I ask questions?
- is it possible to expose TLS settings on Argo Rollout？(My Argo Rollout is ver 1.2.1.)
- Can Prometheus query called in AnalysisTemplate  access Prometheus Ednpoint via HTTPS?

Natsumi

**Comment by NYKYM** [comment]
Feedback on the results of the verification.

- With Argo Rollout, I was able to access the Prometheus Endpoint via https. (*Environment: Argo Rollout (v1.2.1+51c874c), OpenShift 4.10)
- I defined Analysis Template with reference to the following
  - https://argoproj.github.io/argo-rollouts/features/analysis/#analysis-template-arguments
  - Specified content
    - Specify Thanos Querier API Endpoint (https) and Query in provider.web.url
    - Specify Token in provider.web.headers
    - Specify that the result of the query is determined using successCondition and jsonPath
- URL, TOKEN, and query of Thanos Querier were specified referring to the following
  - https://access.redhat.com/solutions/5151831
